src/settings/lib/RuleEditor/rules-show-hint.js

```diff
@@ -328,31 +275,33 @@
       cm.on('focus', this.onFocus = function () { clearTimeout(closingOnBlur); });
     }
 
-    cm.on('scroll', this.onScroll = function () {
+    const startScroll = cm.getScrollInfo();
+
+    cm.on('scroll', this.onScroll = () => {
       const curScroll = cm.getScrollInfo();
       const editor = cm.getWrapperElement().getBoundingClientRect();
-      const newTop = top + startScroll.top - curScroll.top;
+      const newTop = this.position.top + startScroll.top - curScroll.top;
       let point = newTop - (window.pageYOffset || (document.documentElement || document.body).scrollTop);
 
-      if (!below) point += container.offsetHeight;
+      if (!this.isBelow) point += this.container.offsetHeight;
 
       if (point <= editor.top || point >= editor.bottom) return completion.close();
 
-      container.style.top = newTop + 'px';
-      container.style.left = (left + startScroll.left - curScroll.left) + 'px';
+      this.container.style.top = newTop + 'px';
+      this.container.style.left = (this.position.left + startScroll.left - curScroll.left) + 'px';
     });
 
-    CodeMirror.on(hints, 'dblclick', (e) => {
-      const t = getHintElement(hints, e.target || e.srcElement);
+    CodeMirror.on(this.container, 'dblclick', (e) => {
+      const t = getHintElement(this.sections[this.currentSectionIndex].listContainer, e.target || e.srcElement);
```

Comment:
I suggest removing `e.srcElement` check since it is supposed for old versions of IE and we do not use it all over the application and usually refer only to `e.target`.